src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp

```diff
@@ -677,6 +690,17 @@ namespace OpenMS
       results_part[hit_idx].setObservedRT(feature.getRT());
       results_part[hit_idx].setSourceFeatureIndex(feature_index);
       results_part[hit_idx].setObservedIntensity(feature.getIntensity());
+      
+      std::vector<double> mti;
+      for (Size i=0; i<3; ++i)
+      {
+        if (feature.metaValueExists("masstrace_intensity_" + String(i)))
```

Comment:
there is no guarantee that there are 3 isotopes, but thats all one 
usually wants to see. If they are missing, they are reported as NULL.

And the Meta-value is introduced in FFMetabo.
This code is downstream of that tool, and also there is no member in a 
feature which allows to store trace intensities (except to use the 
points in the convex hull; but one cannot know at this point what 
algorithm was used by FFMetabo to compute the intensity from a mass 
trace consisting of many points) - so I guess meta value will have to 
suffice.

Am 3/31/2015 um 3:37 PM schrieb Hannes Roest:
> In src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp
> <https://github.com/OpenMS/OpenMS/pull/1329#discussion_r27478780>:
>
>> @@ -677,6 +690,17 @@ namespace OpenMS
>>        results_part[hit_idx].setObservedRT(feature.getRT());
>>        results_part[hit_idx].setSourceFeatureIndex(feature_index);
>>        results_part[hit_idx].setObservedIntensity(feature.getIntensity());
>> +
>> +      std::vector<double> mti;
>> +      for (Size i=0; i<3; ++i)
>> +      {
>> +        if (feature.metaValueExists("masstrace_intensity_" + String(i)))
>
> sorry, I dont quite understand how you know that there will be 3 values.
> Are these isotopes and is this the aggregate intensity across all
> retention time points?
>
> also it seems somewhat an abuse to store this in a meta value ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenMS/OpenMS/pull/1329/files#r27478780>.
>